src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java

```diff
@@ -474,28 +468,33 @@ private float convertRowToArrow(
             DataValidationUtil.checkValueInRange(inputAsBigDecimal, columnScale, columnPrecision);
 
             if (columnScale != 0 || physicalType == ColumnPhysicalType.SB16) {
+              parsedValue = inputAsBigDecimal;
               ((DecimalVector) vector).setSafe(curRowIndex, inputAsBigDecimal);
               stats.addIntValue(inputAsBigDecimal.unscaledValue());
               rowBufferSize += 16;
             } else {
               switch (physicalType) {
                 case SB1:
+                  parsedValue = inputAsBigDecimal.byteValueExact();
```

Comment:
This wouldn't detect `null` because a `NullPointerException` would be thrown if `validateAndParseBigDecimal` returned null.

Do we need to carry over the `parsedValue` for all the types? There are many types and this can only occur for VARIANTs. Wouldn't it be easier to only have a boolean like `nullDetected` initially as false and set it to true only in the VARIANT case for empty string? WDYT?

Reply:
I think nullDetected makes a lot of sense and would be easier. The reason I added parsedValue to every single value is to have same kind of parallelism between different logical types and also parquetParser but it turns for it complicated the logic.

IIUC, it is the same call as the vector.setSafe() which should make it null safe but regardless i ll go with the nulldetected approach